compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2648,6 +2651,24 @@ public void exitSimpleLiteral(BallerinaParser.SimpleLiteralContext ctx) {
         } else if ((node = ctx.QuotedStringLiteral()) != null) {
             String text = node.getText();
             text = text.substring(1, text.length() - 1);
+            String originalText = text; // to log the errors
+            Pattern pattern = Pattern.compile(Constants.UNICODE_REGEX);
```

Comment:
Since this will be called more than once (in a general scenario when compiling a Ballerina program), I think it is better to compile this Regex once (declare as a top level variable) to reduce the overhead. WDYT?

Reply:
Done